Step 0 to 5 Issues

# Step 01

<img width="1024" alt="Screenshot 2024-11-28 at 21 43 55" src="https://github.com/user-attachments/assets/e42d0d8c-c26a-4f06-beb0-4701b0aa1e59">

`docker compose down --local` 명령어를 실행하니 `--local` 옵션을 찾을 수 없다고 합니다. 
`docker compose down` 명령어로도 문제없이 실행되어 `--local` 옵션을 제외한 후 실습을 진행했습니다. 

# Step 03

<img width="1047" alt="Screenshot 2024-11-28 at 21 51 17" src="https://github.com/user-attachments/assets/9971e657-6941-434c-a7ee-ae42b4776757">


단순히 복사-붙여넣기, 변수(파일)이름 바꾸기로 충분했던 다른 도전과제와 달리, 이 도전과제는 스스로 고민해보고 코드를 수정해야 했습니다. 
이 점에 대해 간단히라도 언급해주는 것이 좋을 것 같습니다. 
저는 아래의 코드로 테스트를 작성했습니다. 

```cs
// tbody 내부 첫 td의 내용은 다음날의 날자가 yyyy-mm-dd 형식이어야 한다. 
var tomorrow = DateTime.Now.AddDays(1).ToString("yyyy-MM-dd");
first.Should().Be(tomorrow);

// tbody 내부 마지막 td의 내용은 날씨 정보이므로 문자열 타입이어야 한다. 
last.Should().BeOfType<string>();
```
# Step 05

<img width="1044" alt="Screenshot 2024-11-28 at 21 58 34" src="https://github.com/user-attachments/assets/ff423f7b-1046-4344-b8db-e038ee5eeefb">

위 도전과제를 진행하던 중 에러가 발생하였습니다. 


테스트 결과:  
eShopLite.WeatherApi.Tests 7 오류와 2 경고와 함께 실패 (12.3초)  
```
#11 8.259 /usr/share/dotnet/sdk/9.0.100/Microsoft.Common.CurrentVersion.targets(2175,5): warning MSB9008: The referenced project '../ServiceDefaults/ServiceDefaults.csproj' does not exist. [/source/WeatherApi/WeatherApi.csproj]
#11 9.142 /source/WeatherApi/Program.cs(4,9): error CS1061: 'WebApplicationBuilder' does not contain a definition for 'AddServiceDefaults' and no accessible extension method 'AddServiceDefaults' accepting a first argument of type 'WebApplicationBuilder' could be found (are you missing a using directive or an assembly reference?) [/source/WeatherApi/WeatherApi.csproj]
#11 9.142 /source/WeatherApi/Program.cs(30,5): error CS1061: 'WebApplication' does not contain a definition for 'MapDefaultEndpoints' and no accessible extension method 'MapDefaultEndpoints' accepting a first argument of type 'WebApplication' could be found (are you missing a using directive or an assembly reference?) [/source/WeatherApi/WeatherApi.csproj]
EXEC : #11 error : process "/bin/sh -c dotnet publish -c Release -o /app" did not complete successfully: exit code: 1
```

eShopLite.ProductApi.Tests 7 오류와 2 경고와 함께 실패 (17.7초)  
```
#15 12.45 /usr/share/dotnet/sdk/9.0.100/Microsoft.Common.CurrentVersion.targets(2175,5): warning MSB9008: The referenced project '../ServiceDefaults/ServiceDefaults.csproj' does not exist. [/source/ProductApi/ProductApi.csproj]
#15 13.05 /source/ProductApi/Program.cs(8,9): error CS1061: 'WebApplicationBuilder' does not contain a definition for 'AddServiceDefaults' and no accessible extension method 'AddServiceDefaults' accepting a first argument of type 'WebApplicationBuilder' could be found (are you missing a using directive or an assembly reference?) [/source/ProductApi/ProductApi.csproj]
#15 13.05 /source/ProductApi/Program.cs(39,5): error CS1061: 'WebApplication' does not contain a definition for 'MapDefaultEndpoints' and no accessible extension method 'MapDefaultEndpoints' accepting a first argument of type 'WebApplication' could be found (are you missing a using directive or an assembly reference?) [/source/ProductApi/ProductApi.csproj]
EXEC : #15 error : process "/bin/sh -c dotnet publish -c Release -o /app" did not complete successfully: exit code: 1
```

eShopLite.WebApp.Tests 7 오류와 2 경고와 함께 실패 (17.9초)  
```
#11 12.45 /usr/share/dotnet/sdk/9.0.100/Microsoft.Common.CurrentVersion.targets(2175,5): warning MSB9008: The referenced project '../ServiceDefaults/ServiceDefaults.csproj' does not exist. [/source/WebApp/WebApp.csproj]
#11 13.05 /source/WebApp/Program.cs(8,9): error CS1061: 'WebApplicationBuilder' does not contain a definition for 'AddServiceDefaults' and no accessible extension method 'AddServiceDefaults' accepting a first argument of type 'WebApplicationBuilder' could be found (are you missing a using directive or an assembly reference?) [/source/WebApp/WebApp.csproj]
#11 13.05 /source/WebApp/Program.cs(39,5): error CS1061: 'WebApplication' does not contain a definition for 'MapDefaultEndpoints' and no accessible extension method 'MapDefaultEndpoints' accepting a first argument of type 'WebApplication' could be found (are you missing a using directive or an assembly reference?) [/source/WebApp/WebApp.csproj]
EXEC : error : failed to solve: process "/bin/sh -c dotnet publish -c Release -o /app" did not complete successfully: exit code: 1
```


파일 의존성과 관련된 문제인 것으로 파악됩니다. 

---

- 실습 과정에 코드 파일을 수정하면서 **어떤 부분이 수정되었고, 이렇게 수정한 이유가 무엇인지** 자세한 설명이 적혀있어 수월하게 실습을 진행할 수 있었습니다. 
- 시간이 빌 때 마다 잠깐씩 실습을 진행하여 정확히 얼마나 걸린지는 확인이 어렵지만, 90분 이내에 실습을 완료하기는 쉽지 않을 것으로 보입니다.

--- a/.gitignore
+++ b/.gitignore
@@ -398,6 +398,7 @@ FodyWeavers.xsd
 *.sln.iml
 
 # Project-specific
+.DS_Store
 workshop/
 workshop*/
 Database.*

--- a/docs/step-00.md
+++ b/docs/step-00.md
@@ -25,7 +25,7 @@
     Get-Command dotnet
     ```
 
-   만약 설치하지 않았다면 [.NET SDK 설치 페이지](https://dotnet.microsoft.com/download/dotnet/9.0)에서 최신 버전을 다운로드 받아 설치합니다.
+   `dotnet`을 실행시킬 수 있는 경로가 보이지 않는다면 설치하지 않은 것입니다. 만약 설치하지 않았다면 [.NET SDK 설치 페이지](https://dotnet.microsoft.com/download/dotnet/9.0)에서 최신 버전을 다운로드 받아 설치합니다.
 
 1. 터미널에서 아래 명령어를 실행시켜 현재 설치한 .NET SDK 버전을 확인합니다.
 
@@ -61,7 +61,7 @@
     Get-Command pwsh
     ```
 
-   만약 설치하지 않았다면 [PowerShell 설치 페이지](https://learn.microsoft.com/powershell/scripting/install/installing-powershell)에서 최신 버전을 다운로드 받아 설치합니다.
+   `pwsh`을 실행시킬 수 있는 경로가 보이지 않는다면 설치하지 않은 것입니다. 만약 설치하지 않았다면 [PowerShell 설치 페이지](https://learn.microsoft.com/powershell/scripting/install/installing-powershell)에서 최신 버전을 다운로드 받아 설치합니다.
 
 1. 터미널에서 아래 명령어를 실행시켜 현재 설치한 PowerShell 버전을 확인합니다.
 
@@ -85,7 +85,7 @@
     Get-Command git
     ```
 
-   만약 설치하지 않았다면 [git CLI 설치 페이지](https://git-scm.com/downloads)에서 최신 버전을 다운로드 받아 설치합니다.
+   `git`을 실행시킬 수 있는 경로가 보이지 않는다면 설치하지 않은 것입니다. 만약 설치하지 않았다면 [git CLI 설치 페이지](https://git-scm.com/downloads)에서 최신 버전을 다운로드 받아 설치합니다.
 
 1. 터미널에서 아래 명령어를 실행시켜 현재 설치한 git CLI 버전을 확인합니다.
 
@@ -109,7 +109,7 @@
     Get-Command gh
     ```
 
-   만약 설치하지 않았다면 [GitHub CLI 설치 페이지](https://cli.github.com/)에서 최신 버전을 다운로드 받아 설치합니다.
+   `gh`을 실행시킬 수 있는 경로가 보이지 않는다면 설치하지 않은 것입니다. 만약 설치하지 않았다면 [GitHub CLI 설치 페이지](https://cli.github.com/)에서 최신 버전을 다운로드 받아 설치합니다.
 
 1. 터미널에서 아래 명령어를 실행시켜 현재 설치한 GitHub CLI 버전을 확인합니다.
 
@@ -133,7 +133,7 @@
     Get-Command docker
     ```
 
-   만약 설치하지 않았다면 [Docker Desktop 설치 페이지](https://docs.docker.com/get-started/introduction/get-docker-desktop/)에서 최신 버전을 다운로드 받아 설치합니다.
+   `docker`을 실행시킬 수 있는 경로가 보이지 않는다면 설치하지 않은 것입니다. 만약 설치하지 않았다면 [Docker Desktop 설치 페이지](https://docs.docker.com/get-started/introduction/get-docker-desktop/)에서 최신 버전을 다운로드 받아 설치합니다.
 
 1. 터미널에서 아래 명령어를 실행시켜 현재 설치한 Docker Desktop 버전을 확인합니다.
 
@@ -157,7 +157,7 @@
     Get-Command code
     ```
 
-   만약 설치하지 않았다면 [Visual Studio Code 설치 페이지](https://code.visualstudio.com/)에서 최신 버전을 다운로드 받아 설치합니다.
+   `code`을 실행시킬 수 있는 경로가 보이지 않는다면 설치하지 않은 것입니다. 만약 설치하지 않았다면 [Visual Studio Code 설치 페이지](https://code.visualstudio.com/)에서 최신 버전을 다운로드 받아 설치합니다.
 
 1. 터미널에서 아래 명령어를 실행시켜 현재 설치한 Visual Studio Code 버전을 확인합니다.
 

--- a/docs/step-01.md
+++ b/docs/step-01.md
@@ -57,12 +57,19 @@ eShopLite
 
     ```bash
     # Terminal 1
+    cd $REPOSITORY_ROOT/workshop
     dotnet run --project ./src/eShopLite.WeatherApi
+    ```
 
+    ```bash
     # Terminal 2
+    cd $REPOSITORY_ROOT/workshop
     dotnet run --project ./src/eShopLite.ProductApi
+    ```
 
+    ```bash
     # Terminal 3
+    cd $REPOSITORY_ROOT/workshop
     dotnet watch run --project ./src/eShopLite.WebApp
     ```
 
@@ -87,6 +94,8 @@ eShopLite
 ```
 
 따라서, 개별 애플리케이션마다 독립적인 컨테이너 이미지를 생성할 때 이 의존성을 해결해야 합니다.
+
+> 이 시점부터는 Docker Desktop 애플리케이션이 실행중이어야 합니다. 작업표시줄에 Docker 아이콘이 보이지 않는다면 Docker Desktop을 실행해주세요.
 
 ### `Dockerfile` 생성: `eShopLite.WebApp`
 
@@ -451,8 +460,8 @@ eShopLite
 1. 터미널 창에서 `Ctrl`+`C`를 눌러 애플리케이션을 종료합니다.
 1. 아래 명령어를 실행시켜 컨테이너와 네트워크, 컨테이너 이미지를 한 번에 삭제합니다.
 
-    ```yml
-    docker compose down --local
+    ```bash
+    docker compose down --rmi local
     ```
 
 ---

--- a/docs/step-03.md
+++ b/docs/step-03.md
@@ -114,16 +114,16 @@ eShopLite
     ```bash
     # Bash/Zsh
     mkdir -p $REPOSITORY_ROOT/workshop/test/eShopLite.WebApp.Tests/Components/Pages
-    touch $REPOSITORY_ROOT/workshop/test/eShopLite.ProductApi.Tests/Components/Pages/ProductsPageTests.cs
+    touch $REPOSITORY_ROOT/workshop/test/eShopLite.WebApp.Tests/Components/Pages/ProductsPageTests.cs
     ```
 
     ```powershell
     # PowerShell
     New-Item -Type Directory -Path $REPOSITORY_ROOT/workshop/test/eShopLite.WebApp.Tests/Components/Pages -Force
-    New-Item -Type File -Path $REPOSITORY_ROOT/workshop/test/eShopLite.ProductApi.Tests/Components/Pages/ProductsPageTests.cs -Force
+    New-Item -Type File -Path $REPOSITORY_ROOT/workshop/test/eShopLite.WebApp.Tests/Components/Pages/ProductsPageTests.cs -Force
     ```
 
-1. `test/eShopLite.ProductApi.Tests/Components/Pages/ProductsPageTests.cs` 파일을 열고 아래와 같이 입력합니다.
+1. `test/eShopLite.WebApp.Tests/Components/Pages/ProductsPageTests.cs` 파일을 열고 아래와 같이 입력합니다.
 
     ```csharp
     using DotNet.Testcontainers.Builders;

